### PR TITLE
`ldrs` Local Disordered Region Sampling and `resre` Renaming Residues Client

### PR DESCRIPTION
**Pull request by menoliu**
Building folded domains by providing the folded domain region location `-fldr` and custom database containing structures that represent the folded domain `-flds` as part of the `cli_build.py` module.

### CODE REVIEW

**Comment by joaomcteixeira** [comment]
The most critical part of this implementation is the sort out the clashes in the chain interface between the folded and the unfolded tails. That was my experience when I first tried it some time ago. You can use the same data structures we have in `build` to accommodate the folded domain, IMHO. So that the pipeline remains almost unchanged. Masks would be needed to avoid calculated clashes (energy function) for the folded domain (if keeping the same pipeline).

Unless there's a better way :-)

:rocket:

**Review by joaomcteixeira — COMMENTED** [review]
Great work! Please provide an example of how to use it in the examples folder.
Many thanks

**Comment by menoliu** [comment]
Development is underway! I will try different strategies I have written in the comments section. Fingers crossed these ideas work. Internal method for building will still work though as I'll be developing in `coords_builder.py`.

File: example/4ebp2_example/README.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+will find the FASTA sequence: ``4ebp2.fasta``, and a tarball containing snapshots of the
+folded domain extracted from PDB ID 2MX4: ``2MX4_4ebp2.tar``.
+
+Steps from now will assume you're in the working director of ``example/4ebp2_example``
+and have already compiled your preferred reusable IDPConformerGenerator database. For
+instructions on the database, please visit the previous exercise "A Real Case Scenario".
+
+Create your folded domain database using the following commands::
+
+    idpconfgen sscalc 2MX4_4ebp2.tar -n
```
Comment (joaomcteixeira) [inline example/4ebp2_example/README.rst:19]: Remember here to add the `-cmd mkdssp` at leat between brackets.

Comment (menoliu) [reply example/4ebp2_example/README.rst:19]: Aha yes, I need to update the documentation also. Some users were confused with `<DSSP CMD>`

Comment (menoliu) [reply example/4ebp2_example/README.rst:19]: Yes, very user I've interacted with didn't know <DSSP CMD> was optional, I think we should do `-cmd mkdssp` to be safe as DSSP v4.X input is different than our compatible v3.X

**Comment by menoliu** [comment]
[3fe09ac](https://github.com/julie-forman-kay-lab/IDPConformerGenerator/pull/231/commits/3fe09acb419055b479517e7ab38a3ce26c87f7ed) is fully "operational" but it's not very good in the sense that un-physical torsion angles at stitching sites are reproduced. However, I am attempting a new method on the translator and rotator stage by mapping backbone `N, CA, C` atoms of the dummy residue onto the folded domain using a covariance matrix about their centroids and using single variable descent to calculate the rotational matrix. 

In the near future, the `rotator` and `pmover` functions will be obsolete. Before optimizing the `clash_count` function for this new approach, I will see if the torsion angles make sense first.

**Comment by menoliu** [comment]
@joaomcteixeira After a long time I have touched up and finished these two clients. I have extensively tested `fldrs` and `resptm` with different proteins. I've also included a note in the documentation to rename any abnormal residue names in the folded domain with `resptm` prior to using `fldrs` so the automatic detection could work. Have a look and cheers :) On-to the next client `contacts` for complexes ;)

**Comment by joaomcteixeira** [comment]
Many, many, thanks @menoliu 
I will test it and review it in the next few days. Please give me ten days more or less. Sorry for the delays.
Cheers!

**Comment by menoliu** [comment]
I am starting up development again! The work so far above this commit is excellent for generating conformers with N-term IDR and C-term IDR tails automatically. From here on I will be working on closing chain breaks with IDRs.

File: example/slc26a9_example/README.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+to **advanced features** that we describe here. Such features include profiting
+from better parallelization that allows modeling longer IDRs in a shorter time
+(e.g., 221 residues in one or two days). Here, we explain how to use the
+modularity of IDPConformerGenerator to exploit its total capacity for modeling
+IDRs by writing two new Python scripts that import IDPConfGen machinery.
+
+The logic behind the LDRS subclient for modeling and IDR connecting two folded
+domains assumes that we have an N-IDR-like case at the C-terminal region of the
+chain break and a C-IDR-like case at the N-terminal region of the other chain
+break. Thus, when defining the IDR sequence in the fast file given to the `-seq`
```
Comment (joaomcteixeira) [inline example/slc26a9_example/README.rst:61]:
```suggestion
first folded domain and a C-IDR-like case at the N-terminal region of the second
domain. Thus, when defining the IDR sequence in the fast file given to the `-seq`
```

**Review by joaomcteixeira — CHANGES_REQUESTED** [review]
1. Suggested minor changes to the example's `README` files
2. I also suggest adding a few sentences on the computational resources required to run these examples and the expected time because these are examples not easy to run on a desktop (or laptop).
3. I suggest adding the output of these examples, that is, the calculated conformers, on a shared folder in @formankay OneDrive and add to the example tutorials the link where users can download the final result just in case they don't have computational resources to run LDRS and still want to evaluate and see the quality of the conformers generated.
4. The lint issues are minor, I won't address them now to avoid adding my signature to your code, but let me know if you have any questions.
5. I will address the minor broken tests.

File: src/idpconfgen/libs/libfunc.py [hunk 3]
```diff
@@ -313,6 +313,36 @@ def raise_(exception, *ignore, **everything):
     raise exception
 
 
+def find_inbetween(num, a, b):
```
[the remark this answers is not in the document]
Comment (menoliu) [reply src/idpconfgen/libs/libfunc.py]: I just checked and I don't use this function anymore. So I removed it :)

**Pull request merged**